src/Util/BatchRequest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Http\Client\Util;
+
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Http\Client\Exception\BatchException;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Implements sending multiple request for client not supporting parallels requests
+ *
+ * Internally, use the sendRequest method
+ *
+ * Should be used with Http\Client\HttpPsrClient
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+trait BatchRequest
+{
+    /**
+     * {@inheritdoc}
+     */
+    abstract public function sendRequest(RequestInterface $request, array $options = []);
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendRequests(array $requests, array $options = [])
+    {
+        $batchResult    = new BatchResult();
+        $batchException = new BatchException();
+        $batchException->setResult($batchResult);
```

Comment:
So, should i keep the current implementation or do you prefere something like this : 

```php
        $batchResult    = new BatchResult();
        $batchException = null;

        foreach ($requests as $request) {
            try {
                $batchResult->addResponse($request, $this->sendRequest($request, $options));
            } catch (Exception $e) {
                if (null === $batchException) {
                    $batchException = new BatchException();
                    $batchException->setResult($batchResult);
                }

                $batchException->addException($request, $e);
            }
        }

        if (null !== $batchException) {
            throw $batchException;
        }

        return $batchResult;
```

or something like this : 

```php
        $batchResult     = new BatchResult();
        $batchExceptions = new \SplObjectStorage();

        foreach ($requests as $request) {
            try {
                $batchResult->addResponse($request, $this->sendRequest($request, $options));
            } catch (Exception $e) {
                $batchExceptions[$request] = $e;
            }
        }

        if (count($batchExceptions) > 0) {
            $batchException = new BatchException();
            $batchException->setResult($batchResult);

            foreach ($batchExceptions as $request => $exception) {
                $batchException->addException($request, $exception);
            }

            throw $batchException;
        }

        return $batchResult;
```